src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java

```diff
@@ -120,8 +119,12 @@ protected MachineRecipe enchant(BlockMenu menu, ItemStack target, ItemStack ench
     }
 
     private boolean isEnchantable(@Nullable ItemStack item) {
-        // stops endless checks of getByItem for enchanted book stacks.
-        if (item != null && item.getType() != Material.ENCHANTED_BOOK) {
+        if (item == null) {
+            return false;
+        } else if (hasIgnoredLore(item)) {
+            return false;
```

Comment:
This is a rather heavy method call as it invokes `.getItemMeta()` so it should only happen further down.
The book check was put in place to aid performance, changing this order will render this performance improvement useless.

Material checks should always come before ItemMeta checks.